### DISCUSSION TITLE
[功能请求]: 快速换课面板

### DISCUSSION THREAD

**Discussion by DryIce-cc** [ClassIsland/ClassIsland#557] [插件功能请求]
### 检查清单

- [X] 我已经在 [Issues](https://github.com/HelloWRC/ClassIsland/issues?q=) 中找过我要提出的问题，没有找到相同问题的ISSUE。
- [X] 我已经详细阅读过应用的帮助文档，但我想要的功能没有实现。
- [X] 我已确认我的功能尚未在 [dev](https://github.com/HelloWRC/ClassIsland/commits/dev/) 分支实现。

### 背景与动机

学校如果要换较多课，右键换课和到档案编辑换课都不太方便。

### 想要实现或优化的功能

将档案编辑的临时层拆分出来，放在桌面右下角，方便换课。

**Comment by DryIce-cc** [comment]
碎片想法
- [ ] 在时间表中提供节次标识填写，方便对应
- [ ] 提供关闭面板按钮 x
- [ ] 在周轮换课程菜单中，支持快速修改周数
- [ ] 临时自定义课程名称
- [ ] 换课：将一节课拖到另一节课
- [ ] 时间表：“此课程不支持换课”选项

**Comment by jiangyin14** [comment]
我觉得可以用“调课”来描述这个面板，之前的右键可以叫“换课”
这个功能可能还是有用的，但是目前我感觉这个换课功能还是挺好用的，没必要搞

**Comment by DryIce-cc** [comment]
开发进度报告  24/8/01

档案编辑现有的课表 DataGrid 无法实现 ComboBox 选择后实时更新，必须手动点击别的列才能读取数据。
目前认为需要使用 Grid + ComboBox 重新实现此组件，亦方便了分割线的添加。
**未查询有没有现成的控件可用。**
另外，对于档案编辑，应当注意 #182，以及支持 ECS 形态的多周轮换课表。
